### PR TITLE
strip stopwords & punctuation and remove changelog from results

### PR DESCRIPTION
**Pull request by melindaloubser1**
closes https://github.com/RasaHQ/rasa-demo/issues/514

### CODE REVIEW

File: actions/actions.py [hunk 1]
```diff
@@ -691,9 +691,18 @@ def run(self, dispatcher, tracker, domain):
...
+            hits = [
+                hit
+                for hit in alg_res.get("hits")
+                if "Rasa X Changelog " not in hit.get("hierarchy", {}).values()
+                and "Rasa Open Source Change Log "
+                not in hit.get("hierarchy", {}).values()
+            ]
+            if not hits:
+                hits = alg_res.get("hits")
+            doc_list = algolia.get_algolia_link(hits, 0)
```
Comment (melindaloubser1) [inline actions/actions.py:703]: Wasn't sure if this belongs here, or in algolia.py.

File: actions/api/algolia.py [hunk 2]
```diff
@@ -20,5 +35,6 @@ def get_algolia_link(self, hits: List, index: int):
         return doc_link
 
     def search(self, search_string: Text):
-        res = self.index.search(search_string)
+        search_text = preprocess_search_text(search_string)
+        res = self.index.search(search_text)
```
Comment (melindaloubser1) [inline actions/api/algolia.py:39]: same here, not sure if this should be in the class or in the `run` method of the action

File: actions/actions.py (hunk 1 above)
Comment (erohmensing) [reply actions/actions.py:703]: Hm, i guess it makes sense to keep it here bc it's not algolia specifc, i.e. if we change the names of the rasa changelogs then we have to update it here

Would you mind changing `alg_res` to `algolia_result` here? It took me longer than I should have to interpret what was happening

File: actions/actions.py [hunk 3]
```diff
@@ -691,9 +691,18 @@ def run(self, dispatcher, tracker, domain):
 
         if alg_res and alg_res.get("hits") and len(alg_res.get("hits")) > 0:
             docs_found = True
-            doc_list = algolia.get_algolia_link(alg_res.get("hits"), 0)
+            hits = [
+                hit
+                for hit in alg_res.get("hits")
+                if "Rasa X Changelog " not in hit.get("hierarchy", {}).values()
+                and "Rasa Open Source Change Log "
+                not in hit.get("hierarchy", {}).values()
```
Comment (erohmensing) [inline actions/actions.py:699]: Would it be easier to check the link url for the word `changelog`?

Comment (erohmensing) [reply actions/actions.py:699]: I think that would have to be after we `get_algolia_link` i guess

File: actions/actions.py (hunk 1 above)
Comment (melindaloubser1) [reply actions/actions.py:703]: ah sure

File: actions/actions.py (hunk 3 above)
Comment (melindaloubser1) [reply actions/actions.py:699]: yeah, when I was testing I just checked if any hierarchy level had "change log" or "changelog" in it, but then I thought maybe that could end up funky (and for Rasa X, the compatibility matrix falls under a "Changelog" heading, and that might be a legitimate result sometimes)

File: actions/actions.py [hunk 4]
```diff
@@ -685,16 +685,28 @@ def run(self, dispatcher, tracker, domain):
             last_user_event = get_last_event_for(tracker, "user", skip=2)
             if last_user_event:
                 search_text = last_user_event.get("text")
```
Comment (melindaloubser1) [inline actions/actions.py:687]:
```suggestion
                search_text = last_user_event.get("text")
                algolia_result = algolia.search(search_text)	
        else:	
            algolia_result = algolia.search(search_text)
```

File: actions/actions.py [hunk 5]
```diff
@@ -685,16 +685,28 @@ def run(self, dispatcher, tracker, domain):
...
-                alg_res = algolia.search(search_text)
-        else:
-            alg_res = algolia.search(search_text)
 
-        if alg_res and alg_res.get("hits") and len(alg_res.get("hits")) > 0:
+        algolia_result = algolia.search(search_text)
```
Comment (melindaloubser1) [inline actions/actions.py:689]:
```suggestion
```

File: actions/actions.py (hunk 3 above)
Comment (erohmensing) [reply actions/actions.py:703]: good point!

File: actions/requirements-actions.txt [hunk 6]
```diff
@@ -5,4 +5,5 @@ beautifulsoup4~=4.6.3
 requests~=2.21.0
 geopy~=1.18.1
 pandas~=0.24.1
-algoliasearch>=2.0,<3.0
\ No newline at end of file
+algoliasearch>=2.0,<3.0
+spacy~=2.1.9
```
Comment (erohmensing) [inline actions/requirements-actions.txt:9]: bahh i wish we didn't need to have spacy as a dependency it's so large 🙈 ah well

Comment (melindaloubser1) [reply actions/requirements-actions.txt:9]: We could use a different stopword list if we wanted to get rid of it?

**Pull request merged**